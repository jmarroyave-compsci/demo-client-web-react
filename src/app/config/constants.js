import _package_ from 'package.json';

const Constants = {
	DEBUG : process.env.NEXT_PUBLIC__DEBUG,
	LOCAL : process.env.NEXT_PUBLIC__LOCAL,
	APP : {
		BASE_PATH: process.env.NEXT_PUBLIC__SITE_URL,
		SITE_URL: process.env.NEXT_PUBLIC__SITE_URL,
		VERSION : _package_.version,
		TITLE: process.env.NEXT_PUBLIC__TITLE,
		PAGE_TITLE: process.env.NEXT_PUBLIC__PAGE_TITLE
	},
	SERVERS: {
		DATA: process.env.NEXT_PUBLIC__DATA, 
		GRAPHQL :  process.env.NEXT_PUBLIC__DATA_GRAPHQL,
	},
	DEFAULTS : {
		PAGING_PAGE_SIZE: 10,
	},
	PLUGINS : {
		GOOGLE: {
			ANALYTICS: process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS,
			MAPS: process.env.NEXT_PUBLIC__GOOGLE_MAPS_KEY,
		},
	},
	FEATURES : {
		SEARCH : (process.env.NEXT_PUBLIC__FEATURE_SEARCH == "true"),
	},
}

export default Constants;