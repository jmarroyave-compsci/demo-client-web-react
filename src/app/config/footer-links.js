import _links from './links';

const links = [
  { "name" : "jmarroyave", items: [
    { "name": "blog", "link" : _links.blog },
  ] },
  { "name" : "labs", items: [
    { "name": "stackoverflow", "link" : _links.labs_trends },
    { "name": "mlb", "link" : _links.labs_mlb },
    { "name": "search", "link" : _links.labs_search },
  ] },
  { "name" : "knowledge base", items: [
    { "name": "android", "link" : _links.kb_android },
  ] },
];

export default links;