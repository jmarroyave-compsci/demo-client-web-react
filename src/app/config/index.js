import _package_ from 'package.json';

const Constants = {
	TEST: (process.env.TESTING === "true") ? true : false,
	DEBUG : process.env.NEXT_PUBLIC__DEBUG,
	LOCAL : process.env.NEXT_PUBLIC__LOCAL,
	APP : {
		BASE_PATH: process.env.NEXT_PUBLIC__BASE_PATH,
		SITE_URL: process.env.NEXT_PUBLIC__SITE_URL,
		VERSION : _package_.version,
		TITLE: process.env.NEXT_PUBLIC__TITLE,
		DESCRIPTION: process.env.NEXT_PUBLIC__DESCRIPTION,
		PAGE_TITLE: process.env.NEXT_PUBLIC__PAGE_TITLE,
		HOME_URL: `${process.env.NEXT_PUBLIC__SITE_URL}${process.env.NEXT_PUBLIC__BASE_PATH}`,
	},
	SERVERS: {
		DATA: process.env.NEXT_PUBLIC__SERVER, 
		WEB_SOCKETS: process.env.NEXT_PUBLIC__SERVER, 
		GRAPHQL :  `${process.env.NEXT_PUBLIC__SERVER}/${process.env.NEXT_PUBLIC__SERVER_API_VERSION}/graphql`,
	},
	DEFAULTS : {
		PAGING_PAGE_SIZE: 10,
		IMAGE_PLACEHOLDER: "/img/ph.jpg",
	},
	PLUGINS : {
		GOOGLE: {
			ANALYTICS: process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS,
			MAPS: process.env.NEXT_PUBLIC__GOOGLE_MAPS_KEY,
		},
	},
	FEATURES : {
		SEARCH : (process.env.NEXT_PUBLIC__FEATURE_SEARCH == "true"),
	},
	SOCIAL_NETWORKS: {
		CARD : {
			TITLE : `${process.env.NEXT_PUBLIC__TITLE} v.${_package_.version}`,
			THUMBNAIL_TWITTER: `${process.env.NEXT_PUBLIC__BASE_PATH}${process.env.NEXT_PUBLIC__CARD_THUMBNAIL_TWITTER}`,
		},
		TWITTER: process.env.NEXT_PUBLIC__SOCIAL_NETWORKS_TWITTER,
	}
}

export default Constants;