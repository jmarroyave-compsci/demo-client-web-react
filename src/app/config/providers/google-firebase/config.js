const config = {
  apiKey: "AIzaSyDaNheMMUtsqx8AjiJAWDcfXwAEBYLJUVY",
  authDomain: "labs-f885b.firebaseapp.com",
  projectId: "labs-f885b",
  //projectId: "labs-f885",
  storageBucket: "labs-f885b.appspot.com",
  messagingSenderId: "606766444808",
  appId: "1:606766444808:web:41eec12c129d582727bcbd",
  measurementId: "G-TNBYCW57TS"
};

export default config;