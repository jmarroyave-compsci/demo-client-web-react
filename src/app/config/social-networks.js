const links = [
  /*
	{enable: true, url: 'https://plus.google.com/112618720945759819017', icon: 'google-plus', color: '#d34836'},
	{enable: true, url: 'https://twitter.com/jmarroyave', icon: 'twitter', color: '#00aced'},
	*/
  { enable: true, url: 'https://facebook.com/jmarroyave', icon: 'facebook', color: '#3b5998' },
  { enable: true, url: 'https://www.linkedin.com/in/jmarroyave', icon: 'linkedin', color: '#0077B5' },
  { enable: true, url: 'https://github.com/jmarroyave', icon: 'github', color: '#000' },
  { enable: true, url: 'mailto:jmarroyave.compsci@gmail.com', icon: 'envelope-o', color: '#000' },
];


export default links;