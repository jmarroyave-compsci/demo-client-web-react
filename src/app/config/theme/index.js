import React from 'react';
import { createTheme } from '@mui/material/styles';
import { red } from '@mui/material/colors';

// Create a theme instance.
export const getTheme = ( mode ) => {
  return createTheme({
    palette: {
      mode: mode,
      primary: {
        main: '#556cd6',
      },
      secondary: {
        main: '#19857b',
      },
      error: {
        main: red.A400,
      },
    },
  });
}