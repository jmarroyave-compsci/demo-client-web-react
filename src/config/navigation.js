
class Navigation {
  constructor(){
    this.drawer = {
      show: false,
    };

    this.options = [
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'Home', 
        icon: 'home', 
        route: '/', 
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        separator: true,
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'Movies', 
        icon: 'movie', 
        route: '/movies', 
      },
        {
          showInMenu: false, 
          showInDrawer: false, 
          name: 'Movie', 
          icon: 'movie', 
          route: '/movie/:id/:name', 
        },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'Podcasts', 
        icon: 'podcasts', 
        route: '/podcasts', 
      },
        {
          showInMenu: false, 
          showInDrawer: false, 
          name: 'Podcast', 
          icon: 'podcast', 
          route: '/podcast/:id/:name', 
        },
      {
        showInMenu: true, 
        showInDrawer: true, 
        separator: true,
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'People', 
        icon: 'people', 
        route: '/people', 
      },
        {
          showInMenu: false, 
          showInDrawer: false, 
          name: 'Person', 
          icon: 'person', 
          route: '/person/:id/:name', 
        },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'Awards', 
        route: '/stories/awards/:entity?/:year?/:page?', 
      },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'Stories', 
        route: '/stories', 
      },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'PodcastMusic', 
        route: '/podcasts/stories/music/:page?', 
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        separator: true,
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'About', 
        icon: 'info', 
        route: '/about', 
      },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'Search', 
        icon: 'search', 
        route: '/search/:qry/:page?', 
      },
    ];
  }
}



const nav = new Navigation();
export default nav;