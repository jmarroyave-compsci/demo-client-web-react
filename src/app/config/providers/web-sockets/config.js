import config from 'app/config';

const CONF = {
  SERVER: config.SERVERS.WEB_SOCKETS
};

export default CONF;