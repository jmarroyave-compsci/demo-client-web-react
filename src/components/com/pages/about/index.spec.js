import React from 'react';
import {render, fireEvent, waitFor, screen, prettyDOM} from '@testing-library/react'
import '@testing-library/jest-dom'

import { itMustHaveNoErrors, StateProvider } from "tests/jest/shared"

import TestComponent from '.';
import config from './.config';

describe(`${config.automata.name.toUpperCase()}`, () => {
  beforeEach(() => { 
	}); 
 
  it('compare Page against snapshot', () => {
    const renderer = render(
      <StateProvider state={{}}>
        <TestComponent render="page"/>
      </StateProvider>
    )
   

    itMustHaveNoErrors(renderer)
    expect(renderer.container).toMatchSnapshot();
  });

  it('compare against snapshot', () => {
    const renderer = render(
      <StateProvider state={{}}>
        <TestComponent render="banner"/>
      </StateProvider>
    )

    
    itMustHaveNoErrors(renderer)
    expect(renderer.container).toMatchSnapshot();
  });
});
