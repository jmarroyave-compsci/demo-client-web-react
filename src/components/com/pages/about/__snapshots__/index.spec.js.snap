// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ABOUT compare Page against snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  box-sizing: border-box;
  width: 100%;
  height: inherit;
  border-radius: 0;
  background: linear-gradient(60deg, #1976d2 0%, #000 150%);
  color: #fff;
}

@media (min-width:0px) {
  .emotion-1 {
    padding: 1.5rem 1.5rem 1.5rem 1.5rem;
  }
}

@media (min-width:600px) {
  .emotion-1 {
    padding: 2rem 3rem 2rem 3rem;
  }
}

@media (min-width:900px) {
  .emotion-1 {
    padding: 3rem 2.5rem 3rem 2.5rem;
  }
}

.emotion-3 {
  padding: 0 0 1rem 0;
  margin: 0;
}

.emotion-4 {
  font-size: 1rem;
  margin-bottom: 1rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1rem;
}

@media (min-width:0px) {
  .emotion-5 {
    padding: 0rem 2rem 3rem 2rem;
  }
}

@media (min-width:600px) {
  .emotion-5 {
    padding: 0rem 2rem 3rem 2rem;
  }
}

@media (min-width:1200px) {
  .emotion-5 {
    padding: 0rem 4rem 3rem 4rem;
  }
}

.emotion-6 {
  padding: 0rem 0rem 4rem 0rem;
}

.emotion-8 {
  box-sizing: border-box;
  width: 100%;
  height: inherit;
  border-radius: 0;
  background: inherit;
  color: inherit;
}

@media (min-width:0px) {
  .emotion-8 {
    padding: 1.5rem 1.5rem 3rem 1.5rem;
  }
}

@media (min-width:600px) {
  .emotion-8 {
    padding: 2rem 3rem 4rem 3rem;
  }
}

@media (min-width:900px) {
  .emotion-8 {
    padding: 3rem 2.5rem 2.5rem 2.5rem;
  }
}

.emotion-12 {
  padding-top: 2rem;
}

.emotion-13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.9375rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 7px 21px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(25, 118, 210, 0.5);
  color: #1976d2;
}

.emotion-13::-moz-focus-inner {
  border-style: none;
}

.emotion-13.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-13 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-13:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
  border: 1px solid #1976d2;
}

@media (hover: none) {
  .emotion-13:hover {
    background-color: transparent;
  }
}

.emotion-13.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-14 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-36 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
}

.emotion-37 {
  padding: 0.85rem;
}

.emotion-38 {
  font-size: 1.5rem;
  padding-bottom: 0.5rem;
}

.emotion-39 {
  font-size: 0.9rem;
}

.emotion-40 {
  cursor: pointer;
  color: rgba(0, 0, 0, 0.87);
  background-color: transparent;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-style: dotted;
  text-decoration-color: rgba(0, 0, 0, 0.87);
  text-decoration-thickness: 1px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: inline-block;
  width: 100%;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      style="margin-bottom: 2rem;"
    >
      <div
        class="com-layout-section section-0 MuiBox-root emotion-1"
      >
        <div
          class="emotion-0"
        >
          <h2
            class="emotion-3"
          >
            How was made?
          </h2>
          <div
            class="emotion-4"
          >
            some details about how this project was made
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-5"
    >
      <div
        class="MuiGrid-root page-module MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-9"
        >
          <div
            class="emotion-6"
          >
            <div
              class="emotion-0"
            >
              <div
                class="com-layout-section undefined MuiBox-root emotion-8"
              >
                <div
                  class="emotion-0"
                >
                  <h2
                    class="emotion-3"
                  >
                    Project
                  </h2>
                  <div
                    class="emotion-4"
                  >
                    The story of this project
                  </div>
                  <div
                    class="emotion-12"
                  >
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root emotion-13"
                      tabindex="0"
                      type="button"
                    >
                      See all
                      <span
                        class="MuiTouchRipple-root emotion-14"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="com-layout-section undefined MuiBox-root emotion-8"
              >
                <div
                  class="emotion-0"
                >
                  <h2
                    class="emotion-3"
                  >
                    Data?
                  </h2>
                  <div
                    class="emotion-4"
                  >
                    source of the data used in this project
                  </div>
                  <div
                    class="emotion-12"
                  >
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root emotion-13"
                      tabindex="0"
                      type="button"
                    >
                      See all
                      <span
                        class="MuiTouchRipple-root emotion-14"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="com-layout-section undefined MuiBox-root emotion-8"
              >
                <div
                  class="emotion-0"
                >
                  <h2
                    class="emotion-3"
                  >
                    What about the code?
                  </h2>
                  <div
                    class="emotion-4"
                  >
                    some details about the code, commits, LoC
                  </div>
                  <div
                    class="emotion-12"
                  >
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root emotion-13"
                      tabindex="0"
                      type="button"
                    >
                      See all
                      <span
                        class="MuiTouchRipple-root emotion-14"
                      />
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="com-layout-section undefined MuiBox-root emotion-8"
              >
                <div
                  class="emotion-0"
                >
                  <h2
                    class="emotion-3"
                  >
                    How the project has evolved?
                  </h2>
                  <div
                    class="emotion-4"
                  >
                    a timeline since the project creation
                  </div>
                  <div
                    class="emotion-12"
                  >
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root emotion-13"
                      tabindex="0"
                      type="button"
                    >
                      See all
                      <span
                        class="MuiTouchRipple-root emotion-14"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
        >
          <div
            style="padding-left: 1rem; padding-right: 0.5rem;"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-36"
            >
              <div
                class="emotion-37"
              >
                <div
                  class="emotion-38"
                >
                  See also
                </div>
                <div
                  class="emotion-39"
                >
                  <span
                    class="emotion-40"
                  >
                    Movie Awards
                  </span>
                </div>
                <div
                  class="emotion-39"
                >
                  <span
                    class="emotion-40"
                  >
                    Movie Remakes
                  </span>
                </div>
                <div
                  class="emotion-39"
                >
                  <span
                    class="emotion-40"
                  >
                    Movie Directors
                  </span>
                </div>
                <div
                  class="emotion-39"
                >
                  <span
                    class="emotion-40"
                  >
                    Movie Producers
                  </span>
                </div>
                <div
                  class="emotion-39"
                >
                  <span
                    class="emotion-40"
                  >
                    Movie Actors
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ABOUT compare against snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  padding: 0 0 1rem 0;
  margin: 0;
}

.emotion-2 {
  font-size: 1rem;
  margin-bottom: 1rem;
}

.emotion-3 {
  padding-top: 2rem;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.9375rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 7px 21px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(25, 118, 210, 0.5);
  color: #1976d2;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
  border: 1px solid #1976d2;
}

@media (hover: none) {
  .emotion-4:hover {
    background-color: transparent;
  }
}

.emotion-4.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="emotion-0"
  >
    <h2
      class="emotion-1"
    >
      How was made?
    </h2>
    <div
      class="emotion-2"
    >
      some details about how this project was made
    </div>
    <div
      class="emotion-3"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root emotion-4"
        tabindex="0"
        type="button"
      >
        See all
        <span
          class="MuiTouchRipple-root emotion-5"
        />
      </button>
    </div>
  </div>
</div>
`;
